frontend/src/components/pages/RegistrantExperience/PersonalInfo/ContactCard.tsx

```diff
@@ -0,0 +1,225 @@
+  PersonalInfoReducerDispatch,
+} from "../../../../types/PersonalInfoTypes";
+import { EmergencyContact } from "../../../../types/CamperTypes";
+import RequiredAsterisk from "../../../common/RequiredAsterisk";
+import {
+  checkEmail,
+  checkFirstName,
+  checkLastName,
+  checkPhoneNumber,
+  checkRelationToCamper,
+} from "./personalInfoReducer";
+
+type ContactCardProps = {
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  contact: EmergencyContact;
+  contactId: number;
+  setPersonalInfo: (action: PersonalInfoReducerDispatch) => void;
+};
+
+const ContactCard = ({
+  nextBtnRef,
+  contact,
+  contactId,
+  setPersonalInfo,
+}: ContactCardProps): React.ReactElement => {
+  const [isFirstNameInvalid, setIsFirstNameInvalid] = useState<boolean>(false);
+  const [isLastNameInvalid, setIsLastNameInvalid] = useState<boolean>(false);
+  const [isEmailInvalid, setIsEmailInvalid] = useState<boolean>(false);
+  const [isPhoneNumberInvalid, setIsPhoneNumberInvalid] = useState<boolean>(
+    false,
+  );
+  const [isRelationInvalid, setIsRelationInvalid] = useState<boolean>(false);
+
+  useEffect(() => {
+    const updateFormErrorMsgs = () => {
+      console.log(
+        "contact",
+        checkFirstName(contact.firstName),
+        checkLastName(contact.lastName),
+        checkEmail(contact.email),
+      );
+      if (!checkFirstName(contact.firstName)) setIsFirstNameInvalid(true);
+      if (!checkLastName(contact.lastName)) setIsLastNameInvalid(true);
+      if (!checkEmail(contact.email)) setIsEmailInvalid(true);
+      if (!checkPhoneNumber(contact.phoneNumber)) setIsPhoneNumberInvalid(true);
+      if (!checkRelationToCamper(contact.relationshipToCamper))
+        setIsRelationInvalid(true);
+    };
+
+    if (nextBtnRef && nextBtnRef.current) {
+      // Passing the same reference
+      nextBtnRef.current.addEventListener("click", updateFormErrorMsgs);
+    }
+
+    return () => {
+      // Passing the same reference
+      if (nextBtnRef && nextBtnRef.current) {
+        nextBtnRef.current.removeEventListener("click", updateFormErrorMsgs);
+      }
+    };
+  }, [contact]);
+
+  return (
+    <Box boxShadow="lg" rounded="xl" borderWidth={1} width="100%">
+      <Box backgroundColor="#FFFFFF" rounded="xl">
+        <Heading textStyle="displayLarge">
+          <Text
+            py={6}
+            px={{ sm: "5", lg: "20" }}
+            textStyle={{ sm: "xSmallBold", lg: "displayLarge" }}
+          >
+            {contactId === 0 ? "Primary Contact" : "Secondary Contact"}{" "}
+          </Text>
+        </Heading>
+        <Divider borderColor="border.secondary.100" />
+      </Box>
+
+      <Box px={{ sm: "5", lg: "20" }}>
+        <Wrap pt="7">
+          <WrapItem width={{ sm: "100%", md: "45%", lg: "20%" }}>
+            <FormControl isInvalid={isFirstNameInvalid}>
+              <FormLabel>
+                <Text textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}>
+                  First Name <RequiredAsterisk />
```

Comment:
Only the primary contact has the fields to be required. Could you make it so that the astrix  don't appear for the secondary contact but if the parent starts filling out one of the fields, then all have to be filled?

Reply:
Understood. I'll edit `checkPersonalInfoFilled` by breaking up and updating primary and secondary contact checks.